src/com/cft/contactmerge/contact/tests/NameTests.java

```diff
@@ -180,4 +180,31 @@ void Name_setParseSuffixAndPrefixInLastName_PunctuatedLastName(){
         assertEquals("Mr.", personOne.getPrefix(), testFailedMessage("Mr.", personOne.getPrefix()));
         assertEquals("Jr.", personOne.getSuffix(), testFailedMessage("Jr.", personOne.getSuffix()));
     }
+
+    @Test
+    void Name_isMiddleNameInFullName_True_PunctuationIgnoreSpaces(){
+        Name name = new Name();
+
+        assertTrue(name.isMiddleNameInFullName("Mr.Page,Christopher,James,1st"));
+        assertTrue(name.isMiddleNameInFullName("Mr. Page, Christopher, James, I"));
+        assertTrue(name.isMiddleNameInFullName("Page, Christopher, James"));
+    }
+
+    @Test
+    void Name_isMiddleNameInFullName_True_MissingCommas(){
+        Name name = new Name();
+
+        assertTrue(name.isMiddleNameInFullName("Mr.Page Christopher James 1st"));
+        assertTrue(name.isMiddleNameInFullName("Mr.Page Christopher James"));
+        assertTrue(name.isMiddleNameInFullName("Page Christopher James"));
+    }
+
+    @Test
+    void Name_isMiddleNameInFullName_False_PunctuationIgnoreSpaces(){
+        Name name = new Name();
+
+        assertFalse(name.isMiddleNameInFullName("Mr.Page,James,1st"));
+        assertFalse(name.isMiddleNameInFullName("Mr. Page, James, I"));
+        assertFalse(name.isMiddleNameInFullName("Page, James"));
+    }
```

Comment:
What about undeterminable name combinations where we have multiple first names or multiple last names? We don't have to handle in this version but we should have a TODO indicating that we don't handle those cases yet.

Reply:
It's on the to do list for later, when the only method that would currently use this method... is setFullName(string name).